libdnf/rpm/solv_sack.cpp

```diff
@@ -349,9 +350,60 @@ void SolvSack::Impl::internalize_libsolv_repos() {
     }
 }
 
-void SolvSack::Impl::load_repo(Repo & repo, bool build_cache, LoadRepoFlags flags) {
+bool SolvSack::Impl::load_repo(Repo & repo, bool build_cache, LoadRepoFlags flags) {
+    auto repo_impl = repo.p_impl.get();
+
+    switch (repo_impl->type) {
+        case Repo::Type::SYSTEM:
+            return load_system_repo(repo);
+        case Repo::Type::AVAILABLE:
+            load_available_repo(repo, build_cache, flags);
+            return true;
+        case Repo::Type::COMMANDLINE:
+            // TODO(jrohel): Implement command line repo loading
+            throw LogicError("Not implemented");
+    }
+
+    return false;
+}
+
+bool SolvSack::Impl::load_system_repo(Repo & repo) {
```

Comment:
My question is who and when will create a system or a commandline repo? Will we create them by default, or on request? Or when it will be needed.
If two operation - creation and load - will be performed at the same time we probably have to use a different approach to load them.